### PR TITLE
Add support for DOI to QuickStatements

### PR DESCRIPTION
**Pull request by carlinmack**
Rebased version of #2340

If/when this is merged I will rebase the others :)

Introduces DOI to QuickStatements (using the Crossref API) resulting in renaming the tool to Identifier to QuickStatements. 
    
image

Note that this does not affect the current `/doi/[doi]` tool which creates QuickStatements using Citations.js and has different behaviour.

### Caveats

> Please list anything which has been left out of this PR or which should be considered before this PR is accepted
Check any of the following which apply:

* [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ]  This change requires a documentation update
    * [ ]  I have made corresponding changes to the documentation
* [ ]  This change requires new dependencies (please list)

*If you make changes to the Python code*
  
* [ ]  My code passes the [tox](https://tox.readthedocs.io/en/latest/) check, I can receive warnings about tests, documentation or both

### Testing
Please describe the [tests](https://numpy.org/doc/stable/reference/testing.html) that you ran to verify your changes. Provide instructions, so we can reproduce. Please also list any relevant details for your test configuration.

* I tested using identifiers found in Wikidata and not
* I created a test in test_utils.py to test the changes I made to `string_to_type`

### Checklist
* [ ] I have [commented](https://peps.python.org/pep-0008/#comments) my code, particularly in hard-to-understand areas
* [ ] My changes generate no new warnings
* [x] I have not used code from external sources without attribution
* [x] I have considered [accessibility](https://www.w3.org/standards/webdesign/accessibility) in my implementation 
* [x] There are no remaining debug statements (print, console.log, ...)

### CODE REVIEW

File: scholia/arxiv.py [hunk 1]
```diff
@@ -16,17 +16,14 @@
...
-
 import re
 
 import sys
 
 import requests
-
 from feedparser import parse as parse_api
-
+from flask import current_app
```
Comment (carlinmack) [inline scholia/arxiv.py:26]: Last time you flagged this as a problem https://github.com/WDscholia/scholia/pull/2340#discussion_r1377858509

Can be removed as it only seems like it is for debugging

**Comment by carlinmack** [comment]
If you think the code could be better written, can you please save rewriting until we merge the three following PRs! it makes rebasing so much more complicated

**Comment by fnielsen** [comment]
I get a "NameError: name 'string_to_arxivs' is not defined" with copy and paste of "https://arxiv.org/abs/2403.19856" into the main search interface http://127.0.0.1:8100/search?q=https://arxiv.org/abs/2403.19856

**Comment by fnielsen** [comment]
This http://127.0.0.1:8100/search?q=ilbert,%20S.,%20Fiesler,%20C.%20&%20Zimmer,%20M.%20Soc.%20Media%20Soc.%20https://doi.org/10.1177/20563051211019004%20(2021). results in the same error.

**Comment by fnielsen** [comment]
There is lost of flake8 errors with `flake8 scholia`

**Comment by carlinmack** [comment]
This PR doesn't do anything to the main search interface: I am simply modifying the arxiv to quickstatements tool to be [doi, arxiv] to quickstatements. Therefore I don't really have an answer to your issues you raise

File: scholia/utils.py [hunk 2]
```diff
@@ -83,12 +83,21 @@
...
-    elif search(r'10\.\d{4}', string):
+
+    if search(r'10.\d{4,9}/[^\s]+', string, flags=IGNORECASE):
         return 'doi'
-    else:
-        return 'string'
+
+    if search(
+        r"(arxiv:)?(\d{4}.\d{4,5}|[a-z\-]+(\.[A-Z]{2})?\/\d{7})(v\d+)?",
+        string,
```
[the remark this answers is not in the document]
Comment (carlinmack) [reply scholia/utils.py:95]: ah this is why the regex included ^ and $. Unsure what to do, the string to doi function only expects `\d{4}\.\d+` but that's a subset of valid arxivs

**Comment by carlinmack** [comment]
Besides from the regex issue this is now ready for review again, the identifier to quickstatement tool handles all the query strings you have mentioned

**Comment by fnielsen** [comment]
There is a crash with the following query:

http://127.0.0.1:8100/id-to-quickstatements?query=Zhang%2C+R.%2C+Madumal%2C+P.%2C+Miller%2C+T.%2C+Ehinger%2C+K.A.%2C+Rubinstein%2C+B.I.%3A+Invertible+concept-based+explanations+for+cnn+models+with+non-negative+concept+activation+vectors.+In%3A+Proceedings+of+the+AAAI+Conference+on+Artificial+Intelligence.+vol.+35%2C+pp.+11682%E2%80%9311690+%282021%29.+https%3A%2F%2Fdoi.org%2F10.1609%2Faaai.v35i13.17389

**Comment by fnielsen** [comment]
I have merge this PR, and then but in this bug: #2459

**Comment by fnielsen** [comment]
One more error: https://github.com/WDscholia/scholia/issues/2460

**Comment by fnielsen** [comment]
One of the errors are not closed by https://github.com/WDscholia/scholia/pull/2461

**Comment by fnielsen** [comment]
Third error with this PR: https://github.com/WDscholia/scholia/issues/2462

**Comment by fnielsen** [comment]
Fourth problem: https://github.com/WDscholia/scholia/issues/2463

**Comment by fnielsen** [comment]
Fifth problem: https://github.com/WDscholia/scholia/issues/2464

**Pull request merged**